servlet/src/main/java/com/linecorp/armeria/server/servlet/DefaultServletHttpRequest.java

```diff
@@ -0,0 +1,567 @@
+    }
+
+    @Nullable
+    private Cookie[] decodeCookie() {
+        final String cookieValue = httpRequest.headers().get(HttpHeaderNames.COOKIE);
+        if (cookieValue != null) {
+            final Cookies cookies = com.linecorp.armeria.common.Cookie.fromCookieHeader(cookieValue);
+            return cookies.stream().map(c -> {
+                final Cookie cookie = new Cookie(c.name(), c.value());
+                if (c.domain() != null) {
+                    cookie.setDomain(c.domain());
+                }
+                if (c.path() != null) {
+                    cookie.setPath(c.path());
+                }
+                cookie.setSecure(c.isSecure());
+                cookie.setMaxAge(Ints.saturatedCast(c.maxAge()));
+                cookie.setHttpOnly(c.isHttpOnly());
+                return new Cookie(c.name(), c.value());
+            }).toArray(Cookie[]::new);
+        } else {
+            return null;
+        }
+    }
+
+    @Override
+    @Nullable
+    public Cookie[] getCookies() {
+        return cookies;
+    }
+
+    @Override
+    public long getDateHeader(String name) {
+        requireNonNull(name, "name");
+        final String dateHeader = getHeader(name);
+        if (dateHeader == null) {
+            return -1;
+        }
+        for (DateFormat x : FORMATS_TEMPLATE) {
+            try {
+                return x.parse(dateHeader).getTime();
+            } catch (Exception e) {
+                // ignored.
+            }
+        }
+        throw new IllegalArgumentException("failed to parse a date header: " + dateHeader);
+    }
+
+    @Override
+    @Nullable
+    public String getHeader(String name) {
+        requireNonNull(name, "name");
+        return httpRequest.headers().get(name);
+    }
+
+    @Override
+    public Enumeration<String> getHeaderNames() {
+        return Collections.enumeration(
+                httpRequest.headers().names().stream()
+                           .map(AsciiString::toString).collect(ImmutableList.toImmutableList()));
+    }
+
+    @Override
+    public StringBuffer getRequestURL() {
+        return new StringBuffer(httpRequest.scheme() + "://" + httpRequest.authority() + requestURI);
+    }
+
+    @Override
+    @Nullable
+    public String getPathInfo() {
+        return pathInfo;
+    }
+
+    @Override
+    @Nullable
+    public String getQueryString() {
+        return serviceRequestContext.query();
+    }
+
+    @Override
+    public String getRequestURI() {
+        return requestURI;
+    }
+
+    @Override
+    public String getServletPath() {
+        return servletPath;
+    }
+
+    @Override
+    public HttpSession getSession(boolean create) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public HttpSession getSession() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public String changeSessionId() {
+        throw new UnsupportedOperationException();
+    }
```

Comment:
At least some sort of minimal session support should exist. It's not easy to find a Servlet application that does not rely on this feature.